### PR TITLE
feat: exclude ports by uids

### PR DESCRIPTION
**Pull request by slonka**
Add an ability to exclude tcp / udp traffic to ports by uids. Closes https://github.com/kumahq/kuma-net/issues/84

Example iptable rules (look for lines with 4201 which is test user id):

```
* nat

# Custom Chains:
--new-chain MESH_INBOUND
--new-chain MESH_OUTBOUND
--new-chain MESH_INBOUND_REDIRECT
--new-chain MESH_OUTBOUND_REDIRECT

# Rules:
--append PREROUTING --protocol tcp --jump MESH_INBOUND
--append OUTPUT --protocol udp --destination-port 53 --match owner --uid-owner 4201 --jump RETURN
--append OUTPUT --protocol udp --destination-port 53 --match owner --uid-owner 5678 --jump RETURN
--append OUTPUT --protocol udp --destination-port 53 --jump REDIRECT --to-ports 45541
--append OUTPUT --protocol tcp --jump MESH_OUTBOUND
--append MESH_INBOUND --protocol tcp --jump MESH_INBOUND_REDIRECT
--append MESH_OUTBOUND --source 127.0.0.6/32 --out-interface lo --jump RETURN
--append MESH_OUTBOUND --protocol tcp ! --destination-port 53 --out-interface lo ! --destination 127.0.0.1/32 --match owner --uid-owner 5678 --jump MESH_INBOUND_REDIRECT
--append MESH_OUTBOUND --protocol tcp ! --destination-port 53 --out-interface lo --match owner ! --uid-owner 5678 --jump RETURN
--append MESH_OUTBOUND --match owner --uid-owner 5678 --jump RETURN
--append MESH_OUTBOUND --protocol tcp --destination-port 53 --jump REDIRECT --to-ports 45541
--append MESH_OUTBOUND --destination 127.0.0.1/32 --jump RETURN
--append MESH_OUTBOUND --jump MESH_OUTBOUND_REDIRECT
--append MESH_INBOUND_REDIRECT --protocol tcp --jump REDIRECT --to-ports 15006
--append MESH_OUTBOUND_REDIRECT --protocol tcp --jump REDIRECT --to-ports 15001

COMMIT
```

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits) yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? yes

### CODE REVIEW

File: test/blackbox_tests/outbound_redirect_test.go [hunk 1]
```diff
@@ -282,6 +284,118 @@ var _ = Describe("Outbound IPv4 TCP traffic to any address:port except excluded
 	)
 })
 
+var _ = Describe("Outbound IPv4 TCP traffic to any address:port except ports excluded by uid ones", func() {
```
Comment (slonka) [inline test/blackbox_tests/outbound_redirect_test.go:287]: In this test I call to the `excludedPort` twice, once using a normal user and then using a `4201` uid, once I assert that I get the redirected response and once `excluded`. The mirror test is for ipv6.

File: iptables/parameters/match_owner.go [hunk 2]
```diff
@@ -47,12 +49,25 @@ func uid(id string, negative bool) *OwnerParameter {
...
 func Uid(id string) *OwnerParameter {
 	return uid(id, false)
 }
 
+// UidRangeOrValue matches if the packet socket's file structure (if it has one) is owned by the user
+// with given list of UID values or ranges
+func UidRangeOrValue(uIDsToPorts config.UIDsToPorts) *OwnerParameter {
+	return uidRangeOrValue(uIDsToPorts)
+}
+
```
Comment (bartsmykla) [inline iptables/parameters/match_owner.go:70]: I think we don't have to create the local function if we are not reusing it anywhere. Wdyt?

```suggestion
// Uid matches if the packet socket's file structure (if it has one) is owned by the user
// with given UID
func Uid(id string) *OwnerParameter {
	return uid(id, false)
}

// UidRangeOrValue matches if the packet socket's file structure (if it has one) is owned by the user
// with given list of UID values or ranges
func UidRangeOrValue(uIDsToPorts config.UIDsToPorts) *OwnerParameter {
	return &OwnerParameter{
		flag:  "--uid-owner",
		value: string(uIDsToPorts.UIDs),
	}
}

```

File: iptables/parameters/protocol.go [hunk 3]
```diff
@@ -75,10 +77,22 @@ func destinationPort(port uint16, negative bool) *TcpUdpParameter {
...
+}
+
 func DestinationPort(port uint16) *TcpUdpParameter {
 	return destinationPort(port, false)
 }
 
+func DestinationPortRangeOrValue(uIDsToPorts config.UIDsToPorts) *TcpUdpParameter {
+	return destinationPortRangeOrValue(uIDsToPorts)
+}
+
```
Comment (bartsmykla) [inline iptables/parameters/protocol.go:95]: The same suggestion here

```suggestion
func DestinationPort(port uint16) *TcpUdpParameter {
	return destinationPort(port, false)
}

func DestinationPortRangeOrValue(uIDsToPorts config.UIDsToPorts) *TcpUdpParameter {
	return &TcpUdpParameter{
		long:  "--destination-port",
		short: "--dport",
		value: string(uIDsToPorts.Ports),
	}
}

```

**Review by bartsmykla — APPROVED** [review]
Just few suggestions, but overall lgtm

**Pull request merged**